code/Base/GamemodeSystem/Entities/BaseGamemode.TargetEntities.cs

```diff
@@ -0,0 +1,44 @@
+namespace Sports;
+
+public partial class BaseGamemode
+{
+	public List<Entity> Entities { get; protected set; } = new();
+
+	public IEnumerable<Entity> GetEntitiesWithTag( string tag )
+	{
+		return Entities.Where( x => x.Tags.Has( tag ) );
+	}
+	public IEnumerable<T> GetEntitiesOfType<T>( string tag = null )
+	{
+		if ( tag is not null )
+		{
+			return Entities.OfType<T>();
+		}
+		else
+		{
+			return GetEntitiesWithTag( tag )
+				.OfType<T>();
+		}
+	}
+
+	[Event.Entity.PostSpawn]
+	protected virtual void GatherTargetEntities()
+	{
+		var ents = FindAllByName( Name );
+
+		foreach ( var entity in ents )
+		{
+			if ( entity == this )
+				continue;
+
+			Log.Debug( $"Found gamemode entity for {Name} - {entity.GetType()}" );
+
+			if ( entity is IGamemodeEntity gamemodeEntity )
+			{
+				Entities.Add( entity );
+				gamemodeEntity.Gamemode = this;
+				entity.Parent = this;
```

Comment:
good thinking with parenting them. We'll have to make sure map cleanup works properly along the line by parenting stuff to the game and/or using a custom cleanup filter. But that's just a random sidenote

Reply:
Yeah!